oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/query/IndexName.java

```diff
@@ -231,26 +231,37 @@ public String nextCustomizedName() {
     }
 
     public static boolean isIndexActive(String indexPath, NodeState rootState) {
+        return isIndexActive(indexPath, rootState, 0);
+    }
+
+    private static boolean isIndexActive(String indexPath, NodeState rootState, int recursionDepth) {
         // An index is active if it has a hidden child node that starts with ":oak:mount-",
         // OR if it is an active merged index
-        try {
-            NodeState indexNode = rootState;
-            for (String e : PathUtils.elements(indexPath)) {
-                indexNode = indexNode.getChildNode(e);
+        if (recursionDepth > 50) {
```

Comment:
Maybe the maximum recursion depth should be exposed as an internal configuration property. 50 seems reasonable but maybe in the future we may want to increase it. In the very least, this constant should be defined as a static field in the class, so it has a clear name.

Reply:
I'm not quite sure what the added indirection would help in this case: we only use the setting once. But OK, I'll add a constant.